web/src/main/java/org/akaza/openclinica/control/submit/DownloadAttachedFileServlet.java

```diff
@@ -119,8 +119,11 @@ public void processRequest() throws Exception {
             addPageMessage("File " + filePathName + " " + respage.getString("not_exist"));
             
             // try to use the passed in the existing file
-            //OC-17868
-        	//file = new File(fileName);
+            //OC-17868, remove any possible path traversal, will make sure only download files from defined download folder 
+            if(fileName != null && fileName.indexOf("..") == -1) {

```

Comment:
Can you use the canonical path like you did for the other fixes?

Reply:
changed to use the canonical path